scipy/optimize/_linprog_highs.py

```diff
@@ -0,0 +1,411 @@
+
+    simplex_primal_edge_weight_strategy : int {0, 1}
+        Strategy for simplex primal edge weights:
+        Dantzig / Devex.  Default is ``0``.
+        Corresponds to the following:
+
+            ``0``: `SIMPLEX_PRIMAL_EDGE_WEIGHT_STRATEGY_DANTZIG`
+
+            ``1``: `SIMPLEX_PRIMAL_EDGE_WEIGHT_STRATEGY_DEVEX`
+
+        ``SIMPLEX_PRIMAL_EDGE_WEIGHT_STRATEGY_*`` are defined as in HiGHS.
+
+    simplex_strategy : int {0, 1, 2, 3}
+        Strategy for simplex solver. Default: ``1``.
+        Corresponds to the following:
+
+            ``0``: `SIMPLEX_STRATEGY_MIN`
+
+            ``1``: `SIMPLEX_STRATEGY_DUAL`
+
+            ``2``: `SIMPLEX_STRATEGY_DUAL_TASKS`
+
+            ``3``: `SIMPLEX_STRATEGY_DUAL_MULTI`
+
+        ``SIMPLEX_STRATEGY_*`` are defined as in HiGHS.
+
+    simplex_update_limit : int
+        Limit on the number of simplex UPDATE operations.  Default
+        is ``5000``.
+    unknown_options : dict
+        Optional arguments not used by this particular solver. If
+        ``unknown_options`` is non-empty, a warning is issued listing all
+        unused options.
+
+    Returns
+    -------
+    sol : dict
+        A dictionary consisting of the fields:
+
+            x : 1D array
+                The values of the decision variables that minimizes the
+                objective function while satisfying the constraints.
+            fun : float
+                The optimal value of the objective function ``c @ x``.
+            slack : 1D array
+                The (nominally positive) values of the slack,
+                ``b_ub - A_ub @ x``.
+            con : 1D array
+                The (nominally zero) residuals of the equality constraints,
+                ``b_eq - A_eq @ x``.
+            success : bool
+                ``True`` when the algorithm succeeds in finding an optimal
+                solution.
+            status : int
+                An integer representing the exit status of the algorithm.
+
+                ``0`` : Optimization terminated successfully.
+
+                ``1`` : Iteration or time limit reached.
+
+                ``2`` : Problem appears to be infeasible.
+
+                ``3`` : Problem appears to be unbounded.
+
+                ``4`` : The HiGHS solver ran into a problem.
+
+            nit : int
+                The total number of iterations performed.
+                For ``solver='simplex'``, this includes iterations in all
+                phases. For ``solver='ipm'``, this does not include
+                crossover iterations.
+            crossover_nit : int
+                The number of primal/dual pushes performed during the
+                crossover routine for ``solver='ipm'``.  This is ``0``
+                for ``solver='simplex'``.
+            message : str
+                A string descriptor of the exit status of the algorithm.
+
+    """
+
+    _check_unknown_options(unknown_options)
+
+    # Catch message_level warnings that HiGHS currently misses:
```

Comment:
From here and below a lot of _longer than >80 characters_ PEP8 issues.

Reply:
Did a first pass fix